src/app/api.py

```diff
@@ -1,127 +1,74 @@
     allow_credentials=True,
     allow_methods=["POST", "GET", "OPTIONS", "PUT", "DELETE"],
     allow_headers=["*"],
 )
+app.cache_backend = InMemoryBackend()
 
 
 @app.on_event("startup")
 async def initialize():
-    logger: logging.Logger = create_logger()
-    logger.info("Startup")
+    app.logger = create_logger()
+    app.logger.info("Startup")
     app.manager = await MetadataManager.create()
+    # await app.cache_backend.setup()
 
 
-def _convert_dict_to_output_model(meta: dict, debug: bool = False) -> ExtractorTags:
-    extractor_tags = ExtractorTags()
-    for key in ExtractorTags.__fields__.keys():
-        if key in meta.keys() and VALUES in meta[key]:
-
-            if not debug or TIME_REQUIRED not in meta[key].keys():
-                meta[key][TIME_REQUIRED] = None
-
-            extractor_tags.__setattr__(
-                key,
-                MetadataTags(
-                    values=meta[key][VALUES],
-                    stars=meta[key][STAR_CASE],
-                    time_for_completion=meta[key][TIME_REQUIRED],
-                    explanation=meta[key][EXPLANATION],
-                ),
-            )
-
-    return extractor_tags
+@app.middleware("http")
+async def caching_and_response_time(request: Request, call_next):
+    start_time = time.time()
+    response = await call_next(request)
+    process_time = time.time() - start_time
+    response.headers["X-Process-Time"] = str(round(process_time * 1000, 2) / 1000)
+    return response
 
 
 @app.post(
     "/extract_meta",
     response_model=Output,
     description="The main endpoint for metadata extraction.",
 )
-async def extract_meta(input_data: Input):
-    starting_extraction = get_utc_now()
-
-    allowance = json.loads(input_data.allow_list.json())
-
-    # build a list of extractors that are set to True
-    whitelist = None if input_data.allow_list is None else [k for k, v in allowance.items() if v]
-    bypass_cache = False if input_data.bypass_cache is None else input_data.bypass_cache
-    meta_data: dict = await app.manager.start(
-        message=Message(
-            url=input_data.url,
-            splash_response=input_data.splash_response,
-            whitelist=whitelist,
-            bypass_cache=bypass_cache,
-        )
-    )
-
-    if meta_data:
-        extractor_tags = _convert_dict_to_output_model(meta_data, input_data.debug)
-
-        if MESSAGE_EXCEPTION in meta_data.keys():
-            exception = meta_data[MESSAGE_EXCEPTION]
-        else:
-            exception = ""
-
-    else:
-        extractor_tags = None
-        exception = f"No response from {METADATA_EXTRACTOR}."
-
-    end_time = get_utc_now()
-    out = Output(
-        url=input_data.url,
-        meta=extractor_tags,
-        exception=exception,
-        time_until_complete=end_time - starting_extraction,
-    )
+@cache(
+    expire=24 * 60 * 60 * 28,
+    key=lambda input, request, response: str(input.url) if input.splash_response is None else None,
```

Comment:
Not entirely sure. I think I had to because otherwise the caching backend which directly uses the return value as key may ran into trouble  (for e.g. postgresql). I'd rather go with explicit than implicit here.